codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/protocols/eventstream/ClientEventStreamUnmarshallerGeneratorTest.kt

```diff
@@ -64,4 +74,156 @@ class ClientEventStreamUnmarshallerGeneratorTest {
+                let conf = crate::Config::builder()
+                    .endpoint_url("http://localhost:1234")
+                    .http_client(http_client.clone())
+                    .behavior_version_latest()
+                    .build();
+                let client = crate::Client::from_conf(conf);
+
+                // This is dummy stream to satisfy input validation for the test service
+                let stream = #{futures_util}::stream::iter(vec![Ok(event)]);
+                let mut response = client
+                    .test_stream_op()
+                    .value(EventStreamSender::from(stream))
+                    .send()
+                    .await
+                    .unwrap();
+
+                #{assert_initial_response:W}
+
+                if let Some(event) = response.value.recv().await.unwrap() {
+                    match event {
+                        TestStream::MessageWithString(message_with_string) => {
+                            assert_eq!("hello, world!", message_with_string.data().unwrap())
+                        }
+                        otherwise => panic!("matched on unexpected variant {otherwise:?}"),
+                    }
+                } else {
+                    panic!("should receive at least one frame");
+                }
+                """,
+                "assert_initial_response" to initialResponseAssertion,
+                "capture_request" to RuntimeType.captureRequest(codegenContext.runtimeConfig),
+                "event_stream" to eventStreamGenerator,
+                "futures_util" to CargoDependency.FuturesUtil.toType(),
+                "initial_response_stream" to initialResponseStreamGenerator,
+            )
+        }
+
+        clientIntegrationTest(
+            rpcEventStreamTestCase.inner.model,
+            IntegrationTestParams(service = "test#TestService"),
+        ) { codegenContext, rustCrate ->
+            rustCrate.testModule {
+                rust("##![allow(unused_imports, dead_code)]")
+                writeUnmarshallTestUtil(codegenContext)
+
+                when (rpcEventStreamTestCase.nonEventStreamMember) {
+                    NonEventStreamMemberInOutput.OPTIONAL_SET, NonEventStreamMemberInOutput.REQUIRED -> {
+                        tokioTest("with_initial_response") {
+                            writeTestBody(
+                                codegenContext,
+                                initialResponseStreamGenerator =
+                                    writable {
+                                        val testCase = rpcEventStreamTestCase.inner
+                                        rust(
+                                            """
+                                            let initial_response = msg(
+                                                "event",
+                                                "initial-response",
+                                                ${testCase.eventStreamMessageContentType.dq()},
```

Comment:
What if there is no "initial-response" sent by the server. Will that be an error?

Reply:
No it isn't. With RPC-bound protocols, if an operation output contains event stream, `try_recv_initial` is always called, and if it turns out the first frame is not an "intial-response", it'll be [saved in the buffer](https://github.com/smithy-lang/smithy-rs/blob/7c5136e88ede599b187b35843b6cff57a5ff5a87/rust-runtime/aws-smithy-http/src/event_stream/receiver.rs#L247-L248) and be read the next time `.recv()` is called on the event receiver.